application/src/main/java/org/togetherjava/tjbot/commands/componentids/ComponentIdInteractor.java

```diff
@@ -0,0 +1,92 @@
+package org.togetherjava.tjbot.commands.componentids;
+
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import org.togetherjava.tjbot.commands.SlashCommand;
+
+import java.util.Arrays;
+
+/**
+ * Delegate class for interacting with component IDs. Provides methods to easily generate valid
+ * component IDs as accepted by the system.
+ * <p>
+ * The method {@link #acceptComponentIdGenerator(ComponentIdGenerator)} must be used before
+ * attempting to generate IDs.
+ * <p>
+ * Mostly used by classes implementing {@link org.togetherjava.tjbot.commands.UserInteractor}, to
+ * simplify handling with the IDs.
+ */
+public final class ComponentIdInteractor {
+    private final String name;
+    private ComponentIdGenerator generator;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param name The unique name of the interactor. Requirements for this are documented in
+     *        {@link net.dv8tion.jda.api.interactions.commands.build.Commands#slash(String, String)}.
+     *        After registration of the interactor, the name must not change anymore.
+     */
+    public ComponentIdInteractor(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Gets the name of the interactor.
+     *
+     * @return the name of the interactor
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Must be used before generating component IDs with {@link #generateComponentId(String...)} and
+     * similar.
+     * <p>
+     * It will provide the interactor a component ID generator through this method, which can be
+     * used to generate component IDs, as used for button or selection menus. See
+     * {@link SlashCommand#onSlashCommand(SlashCommandInteractionEvent)} for details on how to use
```

Comment:
In the command rework the explanation gets moved to the generateComponentId method.

So either add a TODO comment / wait for the command rework to update.

Reply:
ill wait, yeah.

Reply:
done, recheck

Reply:
done